shared/src/store/action-reducers/utils/spatial-elements.ts

```diff
@@ -0,0 +1,86 @@
+import type { Material, Patient, Personnel } from '../../../models';
+import type { Position } from '../../../models/utils';
+import { SpatialTree } from '../../../models/utils/spatial-tree';
+import type { ExerciseState } from '../../../state';
+import type { Mutable, UUID } from '../../../utils';
+import { cloneDeepMutable } from '../../../utils';
+import { updateTreatments } from './calculate-treatments';
+import { getElement } from './get-element';
+
+/**
+ * The element types for which a spatial tree exists in the state to improve the performance.
+ * The position of the element must be changed via one of the function in this file.
+ * In addition, the respective functions must be called when an element gets added or removed.
+ */
+export type SpatialElementType = 'materials' | 'patients' | 'personnel';
+
+/**
+ * Adds an element with a position and executes side effects to guarantee the consistency of the state.
+ * Must be called if an element is added to the state
+ */
+export function addElementPosition(
+    state: Mutable<ExerciseState>,
+    elementType: SpatialElementType,
+    element: Mutable<Material | Patient | Personnel>
+) {
+    if (element.position === undefined) {
```

Comment:
should this throw a reducerError?

Reply:
A reducer error should only be thrown if the action is invalid. Else this would be defensive programming (an anti-pattern).

Reply:
But isn't this already defensive programming? The docs comment clearly states that the `element` must have a position.